test/meson.build

```diff
@@ -1,4 +1,5 @@
 gtest_dep = dependency('gtest', main:true, fallback:['gtest', 'gtest_main_dep'], required:false)
+icu_dep = dependency('icu-i18n', static:static_linkage)
```

Comment:
I looked carefully through https://mesonbuild.com/Dependencies.html  and didn't found that a dependency can be associated with a source file. Dependencies always specified to the exe file only.

libicu is required by libzim, so maybe we should just link our tests executable with libzim, as we do it in tools directory:

```
executable('zimpatch', 'zimpatch.cpp',
  dependencies: libzim_dep,
  install: true)
```

Reply:
Extra favicon.png is removed (from 'data/...', leaved in 'test/data/..')

Reply:
icu is used in tools (https://github.com/openzim/zim-tools/blob/master/src/tools.cpp#L302) so it is not a requirement of libzim (in fact, it is but we cannot rely on this).
All binaries that use `removeAccents` need to link with icu. Not only the tests.
So we need to move the icu dependency global and make "real" binaries linking with it.

On the tests side, it is simpler to make all test linking with it (as it is done with other deps).

For now binaries are working because libzim also depend on it (if compiled with xapian). But we should not rely on this side effect.

Reply:
Actually, we don't need libicu and 'removeAccents' function in zim-tools at all.
Seems it was unused copy/paste from libzim. I just removed `removeAccents()` code and test.
Now we don't need a special case for 'tools-test'.

Reply:
`removeAccents()` seems to be unused copy/paste from libzim. So I just removed it.